### PR TITLE
r/iam_service_linked_role - tagging support

### LINKED ISSUE #22057: Support tags for `aws_iam_service_linked_role` resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please support `tags` variable to `aws_iam_service_linked_role` resource. It is available in AWS management console.

### New or Affected Resource(s)

* aws_iam_service_linked_role

### Potential Terraform Configuration

```hcl
resource "aws_iam_service_linked_role" "elasticbeanstalk" {
  aws_service_name = "elasticbeanstalk.amazonaws.com"
  tags = {
    Name = "hello"
  }
}
```

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #22057

Output from acceptance testing:

```
$ make testacc TESTS=TestAccIAMServiceLinkedRole_ PKG=iam
--- PASS: TestAccIAMServiceLinkedRole_disappears (66.79s)
--- PASS: TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc (82.35s)
--- PASS: TestAccIAMServiceLinkedRole_customSuffix (82.50s)
--- PASS: TestAccIAMServiceLinkedRole_basic (83.02s)
--- PASS: TestAccIAMServiceLinkedRole_description (108.35s)
--- PASS: TestAccIAMServiceLinkedRole_tags (138.13s)
```

### CODE REVIEW

File: internal/service/iam/service_linked_role_test.go [hunk 1]
```diff
@@ -88,18 +88,13 @@ func TestAccIAMServiceLinkedRole_basic(t *testing.T) {
...
 					// Remove existing if possible
-					conn := acctest.Provider.Meta().(*conns.AWSClient).IAMConn
-					deletionID, err := tfiam.DeleteServiceLinkedRole(conn, name)
-					if err != nil {
-						t.Fatalf("Error deleting service-linked role %s: %s", name, err)
-					}
-					if deletionID == "" {
-						return
-					}
+					r := tfiam.ResourceServiceLinkedRole()
```
Comment (DrFaust92) [inline internal/service/iam/service_linked_role_test.go:91]: Nice! I was wreaking my head on how to do
this now I see i didn’t pass the right context..

Comment (ewbankkit) [reply internal/service/iam/service_linked_role_test.go:101]: Actually, need to create an ARN - fixed that now.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TESTS=TestAccIAMServiceLinkedRole_ PKG=iam
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/iam/... -v -count 1 -parallel 20 -run='TestAccIAMServiceLinkedRole_' -timeout 180m
=== RUN   TestAccIAMServiceLinkedRole_basic
=== PAUSE TestAccIAMServiceLinkedRole_basic
=== RUN   TestAccIAMServiceLinkedRole_customSuffix
=== PAUSE TestAccIAMServiceLinkedRole_customSuffix
=== RUN   TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc
=== PAUSE TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc
=== RUN   TestAccIAMServiceLinkedRole_description
=== PAUSE TestAccIAMServiceLinkedRole_description
=== RUN   TestAccIAMServiceLinkedRole_tags
=== PAUSE TestAccIAMServiceLinkedRole_tags
=== RUN   TestAccIAMServiceLinkedRole_disappears
=== PAUSE TestAccIAMServiceLinkedRole_disappears
=== CONT  TestAccIAMServiceLinkedRole_basic
=== CONT  TestAccIAMServiceLinkedRole_tags
=== CONT  TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc
=== CONT  TestAccIAMServiceLinkedRole_disappears
=== CONT  TestAccIAMServiceLinkedRole_description
=== CONT  TestAccIAMServiceLinkedRole_customSuffix
--- PASS: TestAccIAMServiceLinkedRole_disappears (23.64s)
--- PASS: TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc (30.59s)
--- PASS: TestAccIAMServiceLinkedRole_customSuffix (30.59s)
--- PASS: TestAccIAMServiceLinkedRole_basic (30.59s)
--- PASS: TestAccIAMServiceLinkedRole_description (34.87s)
--- PASS: TestAccIAMServiceLinkedRole_tags (53.32s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/iam	57.111s
```

##### GovCloud

```console
% make testacc TESTS=TestAccIAMServiceLinkedRole_ PKG=iam                         
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/iam/... -v -count 1 -parallel 20 -run='TestAccIAMServiceLinkedRole_' -timeout 180m
=== RUN   TestAccIAMServiceLinkedRole_basic
=== PAUSE TestAccIAMServiceLinkedRole_basic
=== RUN   TestAccIAMServiceLinkedRole_customSuffix
=== PAUSE TestAccIAMServiceLinkedRole_customSuffix
=== RUN   TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc
=== PAUSE TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc
=== RUN   TestAccIAMServiceLinkedRole_description
=== PAUSE TestAccIAMServiceLinkedRole_description
=== RUN   TestAccIAMServiceLinkedRole_tags
=== PAUSE TestAccIAMServiceLinkedRole_tags
=== RUN   TestAccIAMServiceLinkedRole_disappears
=== PAUSE TestAccIAMServiceLinkedRole_disappears
=== CONT  TestAccIAMServiceLinkedRole_basic
=== CONT  TestAccIAMServiceLinkedRole_tags
=== CONT  TestAccIAMServiceLinkedRole_customSuffix
=== CONT  TestAccIAMServiceLinkedRole_disappears
=== CONT  TestAccIAMServiceLinkedRole_description
=== CONT  TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc
--- PASS: TestAccIAMServiceLinkedRole_disappears (28.06s)
--- PASS: TestAccIAMServiceLinkedRole_CustomSuffix_diffSuppressFunc (34.44s)
--- PASS: TestAccIAMServiceLinkedRole_customSuffix (34.45s)
--- PASS: TestAccIAMServiceLinkedRole_basic (41.29s)
--- PASS: TestAccIAMServiceLinkedRole_description (48.10s)
--- PASS: TestAccIAMServiceLinkedRole_tags (60.64s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/iam	64.154s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**